core-java-modules/core-java-lang-6/src/test/java/com/baeldung/returnfirstnonnull/ReturnFirstNonNullLazyEvaluateUnitTest.java

```diff
@@ -0,0 +1,79 @@
+package com.baeldung.returnfirstnonnull;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+
+public class ReturnFirstNonNullLazyEvaluateUnitTest {
+
+    @Spy
```

Comment:
You can actually remove this annotation, since you are calling `Mockito.spy` directly, which is totally fine.

---

FYI, the annotation is only required when you are using `@ExtendWith(MockitoExtension.class)` on the test class, and then you would only declare a non-final field:
```
@Spy
private LazyEvaluate spy
```

Reply:
@Spy annotation and associated import removed. I have also implemented your other suggestion regarding package private for the test methods